src/components/numbered-code-snippet/numbered-code-snippet.js

```diff
@@ -11,52 +12,94 @@ function getWindow() {
+    /**
+     * A callback that is invoked when the snippet (or a chunk of the snippet) is copied. If `copyRanges`
+     * are provided, the callback is passed the index (0-based) of the chunk that was copied.
+     */
     onCopy: PropTypes.func,
-    /** CSS that styles the highlighted code. The default theme is a [`highlight.js` theme](https://highlightjs.readthedocs.io/en/latest/style-guide.html#defining-a-theme) theme. It is the dark theme used on mapbox.com's installation flow. */
+    /**
+     * CSS that styles the highlighted code.
+     */
     highlightThemeCss: PropTypes.string.isRequired,
-    /** The width of a character in the theme's monospace font, used for indentation. If you use a font or font-size different than the default theme, you may need to change this value. */
+    /**
+     * The width of a character in the theme's monospace font, used for indentation. If you use a font or
+     * font-size different than the default theme, you may need to change this value.
+     */
     characterWidth: PropTypes.number,
-    /** If the first live chunk (from the `copyRanges` prop) is not visible in the snippet given the `maxHeight`, then autoscroll to make sure the live chunk is in view when the page loads. **/
-    scrollToLive: PropTypes.bool
+    /**
+     * Determines if non-copiable lines (when using `copyRanges`) should be collapsed. Default is true.
+     *
+     * If false, a `maxHeight` is defined, and the first live chunk (from the `copyRanges` prop) is not
+     * visible in the snippet given the `maxHeight`, the component will autoscroll to make sure the live
+     * chunk is in view when the page loads.
+     */
+    collapseLines: PropTypes.bool
   };
 
   static defaultProps = {
     characterWidth: 7.225, // Will need to change this if we change font size
-    scrollToLive: true
+    collapseLines: true
   };
 
+  constructor(props) {
+    super(props);
+    this.state = {
+      expanded: false
+    };
+  }
+
   componentDidMount() {
+    /**
+     * After the component mounts, adjust the position of the highlighted lines,
+     * and add an event listener with a debounced version of the same function
+     * whenever the window is resized.
+     */
     this.adjustPositions();
-    getWindow().addEventListener('resize', this.adjustPositions);
+    getWindow().addEventListener('resize', this.adjustPositionsResize);
 
+    /**
+     * Set the syntax highlighting theme.
+     */
     const theme = this.props.highlightThemeCss;
-
-    // Do not load themes that have already been injected.
+    /* Do not load themes that have already been injected. */
     if (injectedThemes.indexOf(theme) !== -1) return;
     injectedThemes.push(theme);
     const doc = getWindow().document;
     this.styleTag = doc.createElement('style');
     this.styleTag.innerHTML = `${theme}`;
     doc.head.appendChild(this.styleTag);
```

Comment:
@colleenmcginnis I think these lines are the culprit and returning too soon for the second snipped instance to scroll. I recommend moving these lines to is own method:

```
  setTheme = () => {
    const theme = this.props.highlightThemeCss;
    /* Do not load themes that have already been injected. */
    if (injectedThemes.indexOf(theme) !== -1) return;
    injectedThemes.push(theme);
    const doc = getWindow().document;
    this.styleTag = doc.createElement('style');
    this.styleTag.innerHTML = `${theme}`;
    doc.head.appendChild(this.styleTag);
  }
```

And then referencing that method here: `this.setTheme()`